### DISCUSSION TITLE
Why can't I filter for certain types of joins? Include query does not allow where

### DISCUSSION THREAD

**Discussion by estormfield** [prisma/prisma#10677] [Q&A] [answered]
For certain types of relationships, I am unable to filter using where clauses. I am struggling to understand if is this due to specific limitations of Prisma or something about the underlying SQL query that does not allow it. For example:

```
model Post {
  updatedByUser            User?                @relation(name: "updatedByUser", fields: [updatedByUser], references: [id])
  deletedAt                      DateTime?
}
model User {
  updatedPosts   Post[]     @relation(name: "updatedByUser")
  deletedAt                      DateTime?
}

const data = await db.post.findMany({
  include: {
    updatedByUser: { where: { deletedAt: null } },
  },
});
```
I see the following error:
![image](https://user-images.githubusercontent.com/23340128/145871837-9527425a-d988-4010-885e-e0d4b6d4c513.png)

**Comment by austincrim** [comment, marked as answer]
Hey @estormfield 👋 !

First off, I made some changes to your schema locally in order to try to recreate your issue. The schema you posted doesn't include the required ID fields.

```prisma
model Post {
  id            Int       @id @default(autoincrement())
  updatedByUser User?     @relation(fields: [userId], references: [id])
  deletedAt     DateTime?
  userId        Int?
}

model User {
  id           Int       @id @default(autoincrement())
  updatedPosts Post[]
  deletedAt    DateTime?
}
``` 

Could you help me understand what you are trying to accomplish with your example query? The `include` option is only used to specify what relation fields you want to bring back in your query result; it is not meant for filtering.

If you wanted to exclude the records where the `updatedByUser` has a null `deletedAt` field, you could try this query:
```ts
const data = await prisma.post.findMany({
    include: {
      updatedByUser: true
    },
    where: {
      updatedByUser: {
        deletedAt: null
      }
    }
  })
```

Let me know if that works!

**Reply by estormfield** [reply]
Hi @austincrim.

Thanks for the reply. In this case, I want to include the user if they have not been deleted (i.e. have a value other than deletedAt: null) and ignore them if they have been deleted. My understanding is that in your query, if the updatedByUser has been deleted, that post would not be returned

The underlying goal is to create a middleware that allows filtering out soft-deleted entities. I've been unable to find a consistent query syntax that works for different types of entities.

**Reply by MarcosSchlupMaxwell** [reply]
Apparently the solution doesn't work as expected. If there is no `updatedByUser` in the example, then no posts will be returned, when in fact it should be returning posts with `updatedByUser` as null (no relationship found).

I'm also concluding that there's no `where` clause in the case of a 1-1 relation (1-many has it).

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
